### PR TITLE
Approvals api requests

### PR DESCRIPTION
**Pull request by brentwatson**
- API requests for Approvals workflows & approval pools requests.
- Testing strategy & implementation for API requests (using a custom okhttp `Interceptor` that loads local fixture data).
- Added `AndroidDtoVersionExistsTest.kt` test that ensures there is always a android DTO for each core DTO.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,24 +4,85 @@
...
-Add jitpack.io to as a repository.  [Instructions](https://jitpack.io/docs/#how-to).
 
-To use the latest build: `compile 'com.github.percolate:percolate-java-sdk:master-SNAPSHOT'`
-To use a build from a specific commit: `compile 'com.github.percolate:percolate-java-sdk:COMMIT_HASH'`
+The SDK is broken up into 3 libraries:
+
+| Component | Description                      |
+|-----------|----------------------------------|
+| core      | Core objects.  Required.         |
+| autu      | TODO.                            |
```
Comment (msya) [inline README.md]: auth?

Comment (brentwatson) [inline README.md]: :heavy_check_mark:

File: api/build.gradle [hunk 2]
```diff
@@ -50,6 +51,11 @@ jacocoTestReport {
 dependencies {
     compile project(':core')
 
+    testCompile 'junit:junit:4.12'
+
+    // Kotlin Standard Library
+    testCompile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```
Comment (msya) [inline api/build.gradle]: Upgrade Kotlin to 1.0?

Comment (brentwatson) [inline api/build.gradle]: :+1: Moved to Kotlin 1.0 in next PR (#8)

**Pull request merged**